src/main/learnocaml_server_main.ml

```diff
@@ -98,6 +133,16 @@ let base_url =
        Mandatory for '$(b,learn-ocaml build)' if the site is not hosted in path '/', \
        which typically occurs for static deployment."
 
+let child_pid =
+  let open Arg in
+  value & opt int (-1) &
+    info ["child-pid"] ~docv:"CHILD_PID" ~doc:
+      "Set the pid of the child process created by $(i,Lwt_unix.fork) when using \
```

Comment:
> why is it really needed ?

There are three different cases that slightly modify the semantics of learn-ocaml-server:
* no `--serve-during-build` initially,
* `--serve-during-build` and execv learn-ocaml-server from the child process
* `--serve-during-build` and execv learn-ocaml-server from the parent process

As learn-ocaml-server is run by an execv call, this CLI option is necessary.

But I agree that over-documenting in the man page it can be confusing!

> This seems like a completely internal functionality:

Yes, this is the only way for learn-ocaml to pass these extra bits of information to learn-ocaml-server.

> as a code comment rather than in the manpage, where `for internal purposes` should suffice

Looks good to me!

> could you use a variant rather than negative integer codes ?

from ocaml side, this is easy, for example with an option type (None | Some 0 | Some n when n > 0)
but from CLI / cmdliner, I wasn't sure it were as easy.
Anyway I understand hardcoding a `-1` looks ugly, so I'll try to find a solution along these lines.

Reply:
Good news, cmdliner is flexible enough so we can use `Arg.(opt (some int) None)` to use a variant type here.